### PR TITLE
make code more Scala 2.13 friendly

### PR DESCRIPTION
**Pull request by pjfanning**
Some dependencies are not yet ready with scala 2.13 jars but these code changes get a lot of the way towards supporting scala 2.13.

Relates to https://github.com/SpinGo/op-rabbit/issues/171

### CODE REVIEW

**Review by timcharper — APPROVED** [review]

**Comment by pjfanning** [comment]
@timcharper I think the jdk7 build fails in travis because sbt 1.2 doesn't appear to support jdk7 - is it necessary to continue jdk7 support?

**Comment by dalmat36** [comment]
@pjfanning so you are able to get the 2.7.x version of Play JSON to work fine with op-rabbit without any changes? 

When I add it as a dependency with the current 2.1.0 version it seems to cause my client to fail connecting.

**Comment by pjfanning** [comment]
@dalmat36 I am using https://github.com/pjfanning/op-rabbit in production. One difference is that my op-rabbit jars are built with play-json 2.7.4.

**Comment by dalmat36** [comment]
Thanks @pjfanning I found my issue... it was unrelated to op-rabbit. play-json 2.7.4 works fine with the current version.

**Comment by danslapman** [comment]
Any news on this?

**Comment by pjfanning** [comment]
@danslapman I published a 2.13 compatible jar - https://github.com/pjfanning/op-rabbit#installation (I don't have commit/publish rights in this project)

**Comment by timcharper** [comment]
Hi, I approved the changes but did you notice the CI was failing? I didn't see what the issue was. I asked Travis to rebuild

**Comment by timcharper** [comment]
jdk7 is not necessary. Let's remove it from the config and get Travis green and we can merge / publish a new artifact

**Comment by pjfanning** [comment]
@timcharper com.spingo#scoped-fixtures_2.13;2.0.0 needs to be published if we want to support a scala 2.13 build -- https://github.com/SpinGo/scoped-fixtures/issues/2

**Comment by timcharper** [comment]
Can you please rebase this against the latest master? And remove CI changes?

**Comment by pjfanning** [comment]
@timcharper looks like circe 0.12 does not support scala 2.11 and older versions of circe do not support scala 2.13.

**Comment by timcharper** [comment]
@pjfanning I think it's probably ok to just drop 2.11 support right now. 2.12 has been out for a really long time and people have had ample time to upgrade.

**Review by timcharper — APPROVED** [review]

File: addons/airbrake/src/main/scala/com/spingo/op_rabbit/AirbrakeLogger.scala [hunk 1]
```diff
@@ -42,30 +46,36 @@ class AirbrakeLogger(appName: String, airbrakeKey: String, environment: String)
...
+          "consumer" -> name)))
 
       projectRoot(appName)
     }
     new AirbrakeNotifier().notify(notice.newNotice())
   } catch {
     case e: Throwable => log.error("Unable to send airbrake notification for error", e)
   }
+
+  private def asJavaMap(map: Map[String, String]): java.util.Map[String, Object] = {
```
Comment (timcharper) [inline addons/airbrake/src/main/scala/com/spingo/op_rabbit/AirbrakeLogger.scala:74]: Why is this method needed? Isn't it sufficient just to call "Map().asJava" ?

**Comment by timcharper** [comment]
I've published scope-fixtures 2.0.0 for Scala 2.13

File: addons/airbrake/src/main/scala/com/spingo/op_rabbit/AirbrakeLogger.scala (hunk 1 above)
Comment (pjfanning) [reply addons/airbrake/src/main/scala/com/spingo/op_rabbit/AirbrakeLogger.scala:74]: asJava was returning Map<String, String> when the Airbrake APIs require Map<String, Object> so I thought it was better to create a function to produce maps of the required type

**Pull request merged**